torch/distributed/_functional_collectives.py

```diff
@@ -379,13 +488,14 @@ def __new__(cls, elem: torch.Tensor):
             device=elem.device, requires_grad=False
         )
         r.elem = elem
+        r.completed = False
```

Comment:
hmmm I think if the ACT is completed, wouldn't it make more sense to just return the underlying tensor to the user instead of still trying to wrap the data with ACT?

Reply:
> just return the underlying tensor to the user instead of still trying to wrap the data with ACT

Yeah we should do that. I will update the PR.

The flag still serves some purposes though:
- Verifying the behavior of ACT in unit tests.
- Reduce the number of unneccessary `wait_tensor` calls being added to graph during tracing (when the ACT is being consumed by multiple ops).